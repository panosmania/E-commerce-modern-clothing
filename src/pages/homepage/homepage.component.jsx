import React, { Profiler } from "react";

import "./homepage.styles.scss";
import Directory from "../../components/directory/directory.component";

export const Homepage = (props) => (
  <div className="homepage">
    <Profiler
      id="Directory"
      onRender={(id, phase, actualDuration) => {
        console.log({
          id,
          phase,
          actualDuration,
        });
      }}
    >
      <Directory />
    </Profiler>
  </div>
);

export default Homepage;

/* 
<div className="homepage">
    <div className="directory-menu">
      <div className="menu-item">
        <div className="content">
          <h1 className="title">HATS</h1>
          <span className="subtitle">SHOP NOW</span>
        </div>
      </div>
      <div className="menu-item">
        <div className="content">
          <h1 className="title">JACKETS</h1>
          <span className="subtitle">SHOP NOW</span>
        </div>
      </div>
      <div className="menu-item">
        <div className="content">
          <h1 className="title">SNEAKERS</h1>
          <span className="subtitle">SHOP NOW</span>
        </div>
      </div>
      <div className="menu-item">
        <div className="content">
          <h1 className="title">WOMENS</h1>
          <span className="subtitle">SHOP NOW</span>
        </div>
      </div>
      <div className="menu-item">
        <div className="content">
          <h1 className="title">MENS</h1>
          <span className="subtitle">SHOP NOW</span>
        </div>
      </div>
    </div>
  </div>
*/
