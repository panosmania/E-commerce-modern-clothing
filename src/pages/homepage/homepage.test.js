import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { Homepage } from "./homepage.component";

describe("Homepage component", () => {
  const wrapper = shallow(<Homepage />);

  it("should render Homepage component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
