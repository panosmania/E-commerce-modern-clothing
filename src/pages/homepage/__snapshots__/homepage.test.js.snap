// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render Homepage component 1`] = `
<div
  className="homepage"
>
  <Profiler
    id="Directory"
    onRender={[Function]}
  >
    <Connect(Directory) />
  </Profiler>
</div>
`;
